sql/core/src/main/scala/org/apache/spark/sql/DataFrameNaFunctions.scala

```diff
@@ -468,12 +477,21 @@ final class DataFrameNaFunctions private[sql](df: DataFrame) {
       s"Unsupported value type ${v.getClass.getName} ($v).")
   }
 
+  private def toAttributes(cols: Seq[String]): Seq[Attribute] = {
+    cols.map(df.col(_).named.toAttribute)
+  }
+
+  private def outputAttributes: Seq[Attribute] = {
+    df.queryExecution.analyzed.output
+  }
+
   /**
    * Returns a new `DataFrame` that replaces null or NaN values in specified
    * numeric, string columns. If a specified column is not a numeric, string
-   * or boolean column it is ignored.
+   * or boolean column it is ignored. If `cols` is empty, fill() is applied to
+   * all the eligible columns.
```

Comment:
really? Looking at the code, if `cols` is empty, seems we do nothing.

Reply:
forgot to update the comment. thanks for catching!